### PR TITLE
fix(cli): reject snapshots when tenant cluster is not running

### PR DESCRIPTION
**Pull request by r0bnet**
**What issue type does this pull request address?**
/kind bugfix

**What does this pull request do? Which issues does it resolve?** (use `resolves #<issue_number>` if possible)
part of ENGCP-847

Skips snapshot creation and reports the correct status when the tenant cluster is not running, instead of attempting the snapshot and failing with a confusing error later on.

**Please provide a short message that should be published in the vcluster release notes**
Fixed an issue where vcluster attempted to create snapshots for a tenant cluster that was not running, instead of failing fast with a clear status-based error.

**What else do we need to know?**
Added unit tests for the new status parsing/validation logic and an e2e test verifying snapshot creation is rejected while the tenant cluster is paused.

### Additional test suites
Additional test suite(s) that will be executed before the mandatory PR suite:

```label-filter
none
```

---

> [!NOTE]
> **Medium Risk**
> Changes snapshot and cert CLI guardrails on lifecycle status; wrong status detection could block legitimate operations, but scope is narrow with unit and e2e tests.
> 
> **Overview**
> **Snapshot create** now fails fast when the tenant cluster is not **Running**, instead of starting a snapshot pod and failing later with a vague error. Only `snapshot create` sets `RequireRunning` on `initSnapshotCommand`; list/get/restore paths are unchanged.
> 
> `initSnapshotCommand` takes an **`initSnapshotOptions`** struct (credentials, standalone, require running). The running check runs **after** the vCluster is resolved and version-checked, but **before** snapshot URL/credential setup.
> 
> Adds **`VCluster.IsRunning()`** in `find` and uses it for cert **check**/**rotate** abort logic (same behavior, shared helper). Standalone host status parsing moves to **`parseSystemdActiveStatus`** with unit tests.
> 
> **E2E:** ordered pause/resume suite adds a case that `snapshot create` on a **paused** cluster errors with “not running” and **Paused** status.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit a4dc8379619f36c43799df1a626e0a5ad9e0fdc0. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: pkg/cli/find/status_test.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+		},
+		{
+			name: "inactive",
+			out:  []byte("inactive\n"),
+			err:  nil,
+			want: StatusUnknown,
+		},
+		{
+			name: "command error",
+			out:  []byte("inactive\n"),
```
[the remark this answers is not in the document]
Comment (r0bnet) [reply pkg/cli/find/status_test.go:35]: Valid feedback, added this edge case test

File: e2e/test_core/lifecycle/test_pause_resume.go [hunk 2]
```diff
@@ -161,6 +161,19 @@ func PauseResumeScaledDownSpec() {
 			})
 		})
 
+		It("should reject snapshot creation while the tenant cluster is paused", func(ctx context.Context) {
+			By("Attempting to snapshot the paused tenant cluster", func() {
+				out, err := runVClusterCmd(ctx, "snapshot", "create", clusterName,
+					"file:///tmp/e2e-paused-snapshot.tar.gz", "-n", namespace)
```
[the remark this answers is not in the document]
Comment (r0bnet) [reply e2e/test_core/lifecycle/test_pause_resume.go:167]: Really a nitpick (the value is used one time only and this command is meant to fail anyway). So I guess it's fine to leave it like is.

File: pkg/cli/snapshot_helm.go [hunk 3]
```diff
@@ -163,6 +181,19 @@ func initSnapshotCommand(
...
+	if vCluster.IsRunning() {
+		return nil
+	}
+
+	status := vCluster.Status
+	if status == "" {
+		// this can happen when the status is not yet known (e.g no pods running yet right after its creation)
+		status = find.StatusUnknown
+	}
+	return fmt.Errorf("cannot snapshot vCluster %q because it is not running (current status: %q)", vCluster.Name, status)
```
[the remark this answers is not in the document]
Comment (r0bnet) [reply pkg/cli/snapshot_helm.go:194]: As per ticket definition (ENGCP-847) it's okay to only do a restore when status is `Running`. No need for further actions.

File: pkg/cli/find/find.go [hunk 4]
```diff
@@ -66,6 +66,11 @@ const (
 	StatusUnknown          Status = "Unknown"
 )
 
+// IsRunning returns true if the status of the VCluster object equals to `Running`
```
[the remark this answers is not in the document]
Comment (r0bnet) [reply pkg/cli/find/find.go]: Fixed

File: pkg/cli/snapshot_helm.go [hunk 5]
```diff
@@ -163,6 +181,19 @@ func initSnapshotCommand(
...
 }
 
+func validateVClusterIsRunning(vCluster *find.VCluster) error {
+	if vCluster.IsRunning() {
+		return nil
+	}
+
+	status := vCluster.Status
+	if status == "" {
+		// this can happen when the status is not yet known (e.g no pods running yet right after its creation)
```
[the remark this answers is not in the document]
Comment (r0bnet) [reply pkg/cli/snapshot_helm.go]: Fixed

**Review by mfranczy — APPROVED** [review]
